lib/Dialect/AIEX/Transforms/AIEDmaToIpu.cpp

```diff
@@ -335,6 +335,37 @@ struct DmaToIpuPattern : OpConversionPattern<IpuDmaMemcpyNdOp> {
   }
 };
 
+/// Convert IpuDmaWaitOp into IpuSyncOp by retrieving the necessary
+/// information from the ShimDMAAllocationOp referenced through the
+/// symbol argument of this op.
+struct DmaWaitToIpuPattern : OpConversionPattern<IpuDmaWaitOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  DmaWaitToIpuPattern(MLIRContext *context, PatternBenefit benefit = 1)
+      : OpConversionPattern(context, benefit) {}
+
+  LogicalResult
+  matchAndRewrite(IpuDmaWaitOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto dev = op->getParentOfType<AIE::DeviceOp>();
```

Comment:
auto

Reply:
Changed it, but I think this would be an acceptable case in upstream as well as the type is clear from `<AIE::DeviceOp>`

Reply:
Eh one reason I prefer the full type is because with these templated members sometimes you actually get a pointer rather a value so it doesn't actually match what's inside `<...>`.